webhooks/pod_webhook.go

```diff
@@ -47,6 +51,38 @@ type PodMutator struct {
 	Log                       logr.Logger
 }
 
+// BackfillPermissions recovers the state of the flagd-kubernetes-sync role binding in the event of upgrade
+func (m *PodMutator) BackfillPermissions(ctx context.Context) {
+	for i := 0; i < 5; i++ {
+		if i == 5 {
```

Comment:
Let's move this out of the for loop. This gives more focus on the operations we attempt to perform

Plus, at this point our flagd deployments are unstable. Should we avoid operator start?

Reply:
This is a timeout allowing for the cache to be set up, which is managed in a separate groutine (unfortunately there is no 'ready' channel to hook into so the loop is required) -  it does not take into account the health of the deployments, and will backfill permissions even if the pod is crashing, i.e. 
```
NAME                             READY   STATUS             RESTARTS     AGE
busybox-curl-2-67d494d8c-4ldw8   1/2     CrashLoopBackOff   1 (2s ago)   8s
```
will still have its service account added as a subject to the role binding if its annotations are valid.

Ive cleaned up this loop, moving the timeout error to a drop out once the loop has finished